x/proof/keeper/proof.go

```diff
@@ -15,6 +15,8 @@ import (
 func (k Keeper) UpsertProof(ctx context.Context, proof types.Proof) {
 	logger := k.Logger().With("method", "UpsertProof")
 
+	// TODO_IMPROVE(@h5law #427): Use the marshal method on the SparseCompactClosestProof
+	// type here instead in order to reduce space stored on chain.
```

Comment:
`proof` here is referring to the protobuf type, not the closest Merkle proof, which is a field on that type. I think instead of this comment here, a comment should be added to proto/poktroll/proof/proof.proto to rename `closest_merkle_proof` to `compact_closest_merkle_proof`.

I think this comment belongs in pkg/relayer/session/proof.go#99, where `relayer.SessionTree#ProveClosest()` is called (the implementation of which you've already updated the godoc comment for in this PR).

Reply:
My bad 😅 I will rebase onto #406 and add these comments in the appropriate places.